### ISSUE TITLE
openimageio fails to build on unstable and latest master

### ISSUE THREAD

**Issue by eduardosm** [NixOS/nixpkgs#117038]
**Describe the bug**

openimageio fails to build on unstable and latest master

**To Reproduce**
```
nix-build -A openimageio
```

[build-log.txt](https://github.com/NixOS/nixpkgs/files/6175844/build-log.txt)

**Notify maintainers**

@goibhniu

**Comment by illustris** [comment]
This also breaks blender

**Comment by illustris** [comment]
https://slacky.eu/forum/viewtopic.php?t=41596
relevant thread

```patch
diff --git a/src/libOpenImageIO/exif.cpp b/src/libOpenImageIO/exif.cpp
index 10b75c21..0287d9c5 100644
--- a/src/libOpenImageIO/exif.cpp
+++ b/src/libOpenImageIO/exif.cpp
@@ -213,6 +213,9 @@ static const EXIF_tag_info exif_tag_table[] = {
 
 
 
+// libtiff > 4.1.0 defines these in tiff.h. For older libtiff, let's define
+// them ourselves.
+#ifndef GPSTAG_VERSIONID
 enum GPSTag {
     GPSTAG_VERSIONID = 0, 
     GPSTAG_LATITUDEREF = 1,  GPSTAG_LATITUDE = 2,
@@ -237,6 +240,7 @@ enum GPSTag {
     GPSTAG_DIFFERENTIAL = 30,
     GPSTAG_HPOSITIONINGERROR = 31
 };
+#endif
 
 static const EXIF_tag_info gps_tag_table[] = {
     { GPSTAG_VERSIONID,		"GPS:VersionID",	TIFF_BYTE, 4 }, 
@@ -270,7 +274,7 @@ static const EXIF_tag_info gps_tag_table[] = {
     { GPSTAG_AREAINFORMATION,	"GPS:AreaInformation",	TIFF_UNDEFINED, 1 },
     { GPSTAG_DATESTAMP,		"GPS:DateStamp",	TIFF_ASCII, 0 },
     { GPSTAG_DIFFERENTIAL,	"GPS:Differential",	TIFF_SHORT, 1 },
-    { GPSTAG_HPOSITIONINGERROR,	"GPS:HPositioningError",TIFF_RATIONAL, 1 },
+    { GPSTAG_GPSHPOSITIONINGERROR,     "GPS:HPositioningError",TIFF_RATIONAL, 1 },
     { -1, NULL }  // signal end of table
 };
 

```

This patch cherry picked from the thread above fixes openimageio 1.8. These changes are already merged upstream, so openimageio2 works if bumped to 2.2.12.0. Blender still doesn't build, need to find out why.

WIP PR:
https://github.com/NixOS/nixpkgs/pull/117128
